app/viewtils/EvidenceFileUploadsHelper.scala

```diff
@@ -16,19 +16,48 @@
 
 package viewtils
 
-import models.{Mode, UserRequest}
 import models.upload.UploadJourney
+import models.{Mode, UserRequest}
 import play.api.i18n.Messages
 import play.twirl.api.Html
+import uk.gov.hmrc.govukfrontend.views.Aliases.{InsetText, Text}
+import uk.gov.hmrc.govukfrontend.views.html.components.GovukInsetText
 import views.html.components.upload.uploadList
 
 import javax.inject.Inject
 
-class EvidenceFileUploadsHelper @Inject()(uploadList: uploadList){
+class EvidenceFileUploadsHelper @Inject()(govukInsetText: GovukInsetText,
+                                          uploadList: uploadList) {
 
   def displayContentForFileUploads(uploadedFiles: Seq[(UploadJourney, Int)], mode: Mode)(implicit messages: Messages, request: UserRequest[_]): Seq[Html] = {
     uploadedFiles.map(uploadWithIndex =>
       uploadList(uploadWithIndex._1.reference, uploadWithIndex._1.uploadDetails.get.fileName, uploadWithIndex._2 + 1, mode)
     )
   }
+
+  def duplicateDocumentInset(uploadedFiles: Seq[(UploadJourney, Int)])(
+    implicit messages: Messages, request: UserRequest[_]): Option[Html] = {
+
+    val mappedDuplicates = uploadedFiles.groupBy(_._1.uploadDetails.map(_.checksum))
+
+    if (!mappedDuplicates.exists(_._2.size > 1)) {
+      None
+    } else if (mappedDuplicates.count(_._2.size > 1) > 1) {
+      val noOfDuplicateDocs = mappedDuplicates.filter(_._2.size > 1).map(_._2.size).sum
+      Some(govukInsetText(InsetText(content = Text(messages("otherReason.uploadList.multipleDuplicateInsetText", noOfDuplicateDocs)))))
+    } else {
+      val duplicates: Seq[(UploadJourney, Int)] = mappedDuplicates.filter(_._2.size > 1).head._2
+      val firstFile: Int = duplicates.head._2 + 1
+      val otherUploads = duplicates.tail
+      val allOtherUploads = otherUploads.map(_._2 + 1)
+      val insetTextMsg = allOtherUploads.size match {
```

Comment:
Just thinking if we could add some types in this method?

Reply:
Just to clarify, do you mean adding types to the cases in the match statement or the val declarations?

Reply:
Just in the val's will do 😄